### PR TITLE
output options with webpack 1 definitions - wip

### PR DESCRIPTION
**Pull request by gambtho**
with @joshq00
used [WebpackOptionsDefaulter](https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsDefaulter.js) as reference

Need more detail for the following options 
* [ ] `"call"`
* [ ] `"hashFunction"`
* [ ] `"hashDigest"`
* [ ] `"hashDigestLength"`

Need to review for changes in webpack 2
* [ ] `"filename"`
* [ ] `"chunkFilename"`
* [ ] `"library"`
* [ ] `"hotUpdateFunction"`
* [ ] `"jsonpFunction"`
* [ ] `"libraryTarget"`
* [ ] `"path"`
* [ ] `"sourceMapFilename"`
* [ ] `"hotUpdateChunkFilename"`
* [ ] `"hotUpdateMainFilename"`
* [ ] `"crossOriginLoading"`
* [ ] `"devtoolLineToLine"`

### CODE REVIEW

**Comment by gambtho** [comment]
whit this commit

grabbed all output options from the code, and copied anything that was present in the webpack 1 documentation

removed options that weren't in the code

**Comment by sokra** [comment]
I think this is too details for a `concept` page. This content should be in a `api` page.

**Comment by joshq00** [comment]
@sokra we had some discussion about what the "concept" pages should contain. There isn't much content, yet, so we kind of winged it.

Can you describe what you're looking for in these pages? Is it
* When/why you'd use each property (e.g. `output.filename`
* What this section is meant for (use `output` to...)
* Is this for contributors? Users?

**Comment by TheLarkInn** [comment]
As a first time webpack users I would like to know the following: 

- [ ] Most common 1-2 examples of using output with different properties. 
- [ ] Use cases for some more common properties: **see below list of ones to not leave here.**
- [ ] Links to other properties ( like `entry` and `output.filename: '[name].chunk.js' )  that work together with  


## DO NOT ADD INFORMATION FOR THE FOLLOWING PROPS TO THIS DOC: 
 "hotUpdateFunction"
 "jsonpFunction"
 "hotUpdateChunkFilename"
 "hotUpdateMainFilename"
 "crossOriginLoading"
 "devtoolLineToLine"
 "call"
 "hashFunction"
 "hashDigest"
 "hashDigestLength"

@bebraw can you think of anything else? I think this appropriately covers what we're trying to do in this doc.

File: content/concepts/output.md [hunk 1]
```diff
@@ -1,12 +1,227 @@
...
 title: Output
 ---
-> output.path
-> output.filename
-> output.publicPath
-> output.chunkFilename
-> output.jsonpFunction
-> …
+
+## Overview
```
Comment (bebraw) [inline content/concepts/output.md]: You can skip `## Overview` header and then bump `###` to `##` below.

File: content/concepts/output.md [hunk 2]
```diff
@@ -1,12 +1,227 @@
...
+
+Options affecting the output of the compilation. `output` options tell Webpack how to write the compiled files to disk. Note, that while there can be multiple `entry` points, only one `output` configuration is specified.
```
Comment (bebraw) [inline content/concepts/output.md]: Better use webpack (lowercase) so it's consistent with the other material.

File: content/concepts/output.md [hunk 3]
```diff
@@ -1,12 +1,227 @@
...
+```
+
+### Options
+
+The following is a list of values you can pass to the `output` property. 
+
+
+### `output.filename`
+
+Specifies the name of each output file on disk. You must **not** specify an absolute path here! The `output.path` option determines the location on disk the files are written. `filename` is used solely for naming the individual files.
```
Comment (bebraw) [inline content/concepts/output.md]: Why not absolute? In my experience absolute path works as well.

**Comment by bebraw** [comment]
Yeah, we'll most likely end up with two docs:

* Concept level doc to describe the idea of **output** on a high level (entry level doc). Focus on explaining the idea and basic use cases.
* API level doc to dig into details. Focus on technical detail.

It's a good start, though.

**Comment by michael-ciniawsky** [comment]
Could we merge this PR? It goes in the direction I was intending to do it, but I would like to edit a few things and add small examples to ```chunkFilename, sourceMapFilename```, how it actually looks like etc, maybe some structural things aswell. It seems unnecessary I PR my changes since most of it is already done here.

**Comment by bebraw** [comment]
@michael-ciniawsky Done. See my [review notes](https://github.com/webpack/webpack.js.org/pull/156/files). There are a couple of smaller things to fix. Also CI needs some TLC.

**Pull request merged**